datalad/core/local/create.py

```diff
@@ -192,9 +198,23 @@ def __call__(
             description=None,
             dataset=None,
             no_annex=False,
+            annex=True,
             fake_dates=False,
             cfg_proc=None
     ):
+        # TODO: introduced with 0.13, remove with 0.14
+        if annex != (not no_annex) and no_annex:
+            # the two mirror options do not agree and the deprecated one is
+            # not at default value
+            lgr.warning("datalad-create's `no_annex` option is deprecated "
+                        "and will be removed in a future release, "
+                        "use the reversed-sign `annex` option instead.")
```

Comment:
I think `warnings.warn(..., DeprecationWarning)` is preferable here for a few reasons, including consistency within our codebase, allowing developers to control `DeprecationWarnings` in the same way as they do in other spots, and avoiding a flood of warnings by default.

Reply:
Will do!